main/src/main/java/com/sedmelluq/discord/lavaplayer/source/nico/NicoAudioTrack.java

```diff
@@ -14,30 +15,31 @@
 import org.apache.http.client.methods.HttpGet;
 import org.apache.http.client.methods.HttpPost;
 import org.apache.http.entity.StringEntity;
-import org.json.JSONArray;
-import org.json.JSONObject;
-import org.json.JSONTokener;
 import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.io.IOException;
 import java.net.URI;
 import java.nio.charset.StandardCharsets;
-import java.util.Timer;
-import java.util.TimerTask;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
 
 /**
  * Audio track that handles processing NicoNico tracks.
  */
 public class NicoAudioTrack extends DelegatedAudioTrack {
     private static final Logger log = LoggerFactory.getLogger(NicoAudioTrack.class);
 
+    private static final ScheduledExecutorService executorService = Executors.newScheduledThreadPool(1);
+
     private final NicoAudioSourceManager sourceManager;
 
     private String heartbeatURL;
 
-    private JSONObject info;
+    private JsonBrowser info;
```

Comment:
Ideally you want to avoid storing entire JSON strings as these can consume a lot of memory. Keep it to specific bits, like you've done with `heartbeatURL`. e.g.
```java
private int heartbeatDelay;
```